tests/std/tests/GH_002334_branchless_clamp/test.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <string>
+
+using namespace std;
+
+int main() {
+    // Optimized cases
+    assert(clamp(2, 4, 7) == 4);
+    assert(clamp(6, 4, 7) == 6);
+    assert(clamp(8, 4, 7) == 7);
+
+    assert(clamp(2, 4, 7, std::less{}) == 4);
+    assert(clamp(6, 4, 7, std::less{}) == 6);
+    assert(clamp(8, 4, 7, std::less{}) == 7);
+
+    assert(clamp(2, 4, 7, std::less<int>{}) == 4);
+    assert(clamp(6, 4, 7, std::less<int>{}) == 6);
+    assert(clamp(8, 4, 7, std::less<int>{}) == 7);
+
+    const int data[100] = {};
+    assert(clamp(data + 22, data + 30, data + 70) == data + 30);
+    assert(clamp(data + 44, data + 30, data + 70) == data + 44);
+    assert(clamp(data + 88, data + 30, data + 70) == data + 70);
+
+    // Generic cases
+    assert(clamp(2, 4, 7, std::less<long long>{}) == 4);
+    assert(clamp(6, 4, 7, std::less<long long>{}) == 6);
+    assert(clamp(8, 4, 7, std::less<long long>{}) == 7);
+
+    assert(clamp<string>("apple", "banana", "coconut") == "banana");
+    assert(clamp<string>("blueberry", "banana", "coconut") == "blueberry");
+    assert(clamp<string>("cucumber", "banana", "coconut") == "coconut");
+
+#if _ITERATOR_DEBUG_LEVEL < 2
+    // make sure we keep early return for generic cases:
+    // either less than min or greater than max input should produce only one comparison
+    // for _ITERATOR_DEBUG_LEVEL == 2 we validate predicate, so make extra comparisons
+
+    int comparisons        = 0;
+    const auto counted_cmp = [&comparisons](auto a, auto b) {
+        ++comparisons;
+        return a < b;
+    };
+
+    assert(clamp(2, 4, 7, counted_cmp) == 4);
+    assert(clamp(6, 4, 7, counted_cmp) == 6);
+    assert(clamp(8, 4, 7, counted_cmp) == 7);
+    assert(comparisons == 5);

```

Comment:
No change requested: While it would be slightly stricter to check `comparisons` after each call, I see that the precise order in which comparisons are performed is unspecified by the Standard, so I agree with your approach here (just count the total number).